synthea-hiv/uploader/resources.py

```diff
@@ -0,0 +1,131 @@
+"""Resources that OpenMRS can upload. This class is only used for OpenMRS."""
+
+import random
+from typing import Dict, List
+import uuid
+
+import idgen
+
+
+class BaseResource:
+  """Base resource. Has a JSON, original_id, and potentially, a new_id."""
+
+  def __init__(self, json: Dict[str, str]):
+    self.json = json
+    self.original_id = self.json['id']
+    self.new_id = None
+
+  def openmrs_convert(self, *args):
```

Comment:
What is the benefit of this polymorphism? It seems to me that we always deal with `Patient`, `Encounter`, and `Observation` objects directly and never use `BaseResource` outside this module. So given the signature of `openmrs_convert` is different for each of those sub-classes, I wonder what is the benefit of defining this base abstract method here?

I would suggest drop this method and to keep `json`, `original_id`, and `new_id` and the functions that act on them; then use this object in `Patient`, `Encounter`, and `Observation` through _composition_ (not inheritance). Treat this second part as "optional" but I would strongly suggest dropping `openmrs_convert` unless there is good reason to keep it.

Reply:
Will switch to using composition